src/proxy/resolve.rs

```diff
@@ -129,25 +136,28 @@ where
     R: Resolution,
     R::Endpoint: fmt::Debug,
     R::Error: Into<Error>,
-    M: rt::Make<R::Endpoint>,
+    M: svc::Service<R::Endpoint>,
+    M::Error: Into<Error>,
 {
     type Key = SocketAddr;
-    type Service = M::Value;
+    type Service = M::Response;
     type Error = Error;
 
     fn poll(&mut self) -> Poll<Change<Self::Key, Self::Service>, Self::Error> {
         loop {
+            if let Async::Ready(Some((addr, svc))) = self.pending.poll().map_err(Into::into)? {
+                self.is_empty.store(false, Ordering::Release);
```

Comment:
If my understanding is correct, this means that we will only consider the `Discover` "ready" when it has actually built at least one service from its discovery updates, rather than considering it ready as soon as an `Add` update is received?

Reply:
Enter my confusion around `is_empty` and what it is intended to convey...

I think you're right about this change in behavior, though, yes.

Reply:
you may be pleased to hear that all the `is_empty` stuff is going away in #268